odl/trafos/wavelet.py

```diff
@@ -101,13 +100,29 @@ def __init__(self, space, wavelet, nlevels, variant, pad_mode='constant',
             ``pywt`` back-end.
         impl : {'pywt'}, optional
             Back-end for the wavelet transform.
+        axes : sequence of ints, optional
+            Axes over which the DWT that created ``coeffs`` was performed.  The
+            default value of ``None`` corresponds to all axes.
+
         """
         if not isinstance(space, DiscreteLp):
             raise TypeError('`space` {!r} is not a `DiscreteLp` instance.'
                             ''.format(space))
 
+        self.__impl, impl_in = str(impl).lower(), impl
+        if self.impl not in _SUPPORTED_WAVELET_IMPLS:
+            raise ValueError("`impl` '{}' not supported".format(impl_in))
+
+        if axes is None:
+            axes = tuple(range(space.ndim))
+        elif np.isscalar(axes):
+            axes = (axes, )
+        elif len(axes) > space.ndim:
+                raise ValueError("Too many axes.")
+        self.axes = tuple(axes)
```

Comment:
No checks that axes is "reasonable"? E.g. contains proper integers.

Reply:
Duplicate axes are also not allowed. Within `__call__` the axes will be checked by `wavedecn`. Would you prefer to replicate the checks highlighted at the link below within the `__init__` function here?:
https://github.com/PyWavelets/pywt/blob/53f9cdba2c7df4a080fb1a5fe81143a4190f898f/pywt/_multilevel.py#L334-L345